src/Clay/Grid.hs

```diff
@@ -1,58 +1,249 @@
-{-# LANGUAGE OverloadedStrings #-}
--- | Partial implementation of <https://alligator.io/css/css-grid-layout-grid-areas grid area CSS API>.
---
--- For instance, you want to generate the following CSS:
---
--- @
--- .grid1 {
---   display: grid;
---   width: max-content;
--- }
---
--- .grid3 {
---   display: grid;
---   width: max-content;
--- }
---
--- \@media (min-width: 40.0rem) {
---   .grid3 {
---     display: grid;
---     grid-template-columns: 1fr 1fr 1fr;
---     grid-gap: 1rem;
---     width: max-content;
---   }
--- }
--- @
---
--- The corresponding clay code:
---
--- @
---  ".grid1" ? do
---    display grid
---    width maxContent
---  ".grid3" ? do
---    display grid
---    width maxContent
---  query M.screen [M.minWidth (rem 40)] $ ".grid3" ? do
---    display grid
---    gridTemplateColumns [fr 1, fr 1, fr 1]
---    gridGap $ rem 1
---    width maxContent
--- @
+{-# LANGUAGE
+    OverloadedStrings
+  , GeneralizedNewtypeDeriving
+  , TypeFamilies
+  , RankNTypes
+  , ExistentialQuantification
+  , PatternSynonyms
+  #-}
+-- | Implementation of <https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout>.
 module Clay.Grid
-( gridGap
-, gridTemplateColumns
-)
-where
+  ( gap
```

Comment:
Are there many definitions in this module that are _not_ exported? If everything is exported, we can also simply not restrict the exports.